third_party/move/tools/move-lint/tests/cases/bool_comparison/sources/bool_comparison.move

```diff
@@ -0,0 +1,19 @@
+module NamedAddr::Detector {
+    const ERROR_NUM: u64 = 2;
+    public fun func1(x: bool) {
+        if (x == true) {};
+        if (x == false) {};
+        if (x != true) {};
+        if (x != false) {};
+        if (x == true || ERROR_NUM == 2) {};
+        if (x == true && x != false) {};
+        if (x) {};
+        if (!x) {};
+        if (true == x) {};
+        if (condition() == true) {};
```

Comment:
Do we handle `true == condition()`. If not, is there a reason to not handle it?

I think other cases that can be handled here are: `!x == true` and similar.

Reply:
refactored